src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -162,31 +176,30 @@ public int setTagWebId(Tag tag) throws JSONException {
       }
       JSONObject requestResponse = new JSONObject(JsonT);
       JSONArray items = new JSONArray();
-      if (requestResponse.has("Items")) {
-        items = requestResponse.getJSONArray("Items");
-      }
+      if (requestResponse.has("Items")) items = requestResponse.getJSONArray("Items");
 
       if (items.length() > 0) {
         // tag exists
-        tag.setWebID(items.getJSONObject(0).getString("WebId"));
+        tagWebID = items.getJSONObject(0).getString("WebId");
       } else {
         // tag does not exist and must be created
+        String payload = PayloadBuilder.buildNewPointBody(tag); // need datapoint? ////
         res =
             RequestHTTPS(
-                url,
-                "Post",
-                postHeaders,
-                buildNewPointBody(tagName, tag.getDataType()),
-                "",
-                responseFilename);
+                url, "Post", OSIsoftConfig.getPostHeaders(), payload, "", responseFilename);
 
         if (res == NO_ERROR) {
           // The WebID is sent back in the headers of the previous post
           // however, there is no mechanism currently to retrieve it so
           // another request must be issued.

```

Comment:
```suggestion
          /* The WebID is sent back in the headers of the previous post
              however, there is no mechanism currently to retrieve it so
              another request must be issued*/.
```
multiline comments should use block comment syntax instead of single comments